File: ordenacao_selecao.py
# primeiro criamos uma funcao que identifica o index que possui o menor valor da lista
def buscaMenor(arr):
    menor = arr[0]
    menor_indice = 0

    for i in range(1, len(arr)):
        # se o item atual do loop for menor que o primeiro item da lista
        # substitui o atual menor valor e o indice correspondente
        if arr[i] < menor:
            menor = arr[i]
            menor_indice = i

    # retorna o indice que contem o menor valor da lista
    return menor_indice

def ordenarSelecao(arr):
    # criamos um novo array onde iremos anexar os valores ordenados
    novoArr = []

    # para cada item do array base
    for i in range(len(arr)):
        # identificamos o indice que contem o menor valor
        menor = buscaMenor(arr)
        # adicionamos ao novo array e removemos o valor da lista base
        # para que seja usado no loop de busca novamente
        novoArr.append(arr.pop(menor))

    # logo apos retornamos o array ordenado de forma crescente
    return novoArr

print(ordenarSelecao([2,3,1,4,5]))

File: ordenacaoSelecao.java
import java.util.ArrayList;

public class ordenacaoSelecao {
    public static void main(String[] args) {
        
        ArrayList<Integer> arr = new ArrayList<>();

        arr.add(3);
        arr.add(2);
        arr.add(1);
        arr.add(4);
        arr.add(5);

        System.out.println(ordenarSelecao(arr));

    }

    public static Integer buscarMenor(ArrayList<Integer> arr){
        Integer menor = arr.get(0);
        Integer menor_indice = 0;

        for (Integer i = 1; i < arr.size(); i++) {
            if (arr.get(i) < menor) {
                menor = arr.get(i);
                menor_indice = i;
            }
        }

        return menor_indice;
    }

    public static ArrayList<Integer> ordenarSelecao(ArrayList<Integer> arr){
        ArrayList<Integer> novaLista = new ArrayList<>();

        for (Integer i = arr.size(); i != 0; i--) {
            int indice = buscarMenor(arr);
            Integer menor = arr.get(indice);
            novaLista.add(menor);
            arr.remove(indice);
        }

        return novaLista;
    }

}
